pkg/glusterfs/controllerserver.go

```diff
@@ -543,11 +543,31 @@ func (cs *ControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		}, nil
 	}
 
+	snapResp, err := cs.createSnapshot(req.GetName(), req.GetSourceVolumeId())
+	if err != nil {
+		return nil, err
+
+	}
+	return &csi.CreateSnapshotResponse{
+		Snapshot: &csi.Snapshot{
+			Id:             snapResp.VolInfo.Name,
+			SourceVolumeId: snapResp.ParentVolName,
+			CreatedAt:      snapResp.CreatedAt.Unix(),
+			SizeBytes:      int64(snapResp.VolInfo.Capacity),
+			Status: &csi.SnapshotStatus{
+				Type: csi.SnapshotStatus_READY,
+			},
+		},
+	}, nil
+}
+
+func (cs *ControllerServer) createSnapshot(name, sourceVolID string) (*api.SnapCreateResp, error) {
```

Comment:
I don't like having both `createSnapshot` and `CreateSnapshot` when they do different things. Let's get a different name for this one... suggestions: `performSnapshot`, `doSnapshot`, `requestAndActivateSnapshot`

Reply:
@JohnStrunk updated PTAL